libshortfin/CMakeLists.txt

```diff
@@ -46,9 +44,98 @@ list(APPEND CMAKE_MODULE_PATH
 include(shortfin_library)
 
 # Dependencies.
-find_package(spdlog)
-find_package(xtensor)
-find_package(IREERuntime)
+
+if(SHORTFIN_BUNDLE_DEPS)
+  ## fmt
+  FetchContent_Declare(
+    fmt
+    GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+    GIT_TAG        e69e5f977d458f2650bb346dadf2ad30c5320281 # 10.2.1 (sync with spdlog)
+  )
+
+  ## spdlog
+  # We build fmt from source instead, because we also use fmt.
+  set(SPDLOG_FMT_EXTERNAL ON)
+  FetchContent_Declare(
+    spdlog
+    GIT_REPOSITORY https://github.com/gabime/spdlog.git
+    GIT_TAG        2d4acf8cc321d7783d8f2e22e17a794c6d0e9450 # v1.14.1
+  )
+
+  ## xtl: required for xtensor
+  FetchContent_Declare(
+    xtl
+    GIT_REPOSITORY https://github.com/xtensor-stack/xtl.git
+    GIT_TAG        a7c1c5444dfc57f76620391af4c94785ff82c8d6 # v0.7.7
+  )
+
+  ## xtensor
+  FetchContent_Declare(
+    xtensor
+    GIT_REPOSITORY https://github.com/xtensor-stack/xtensor.git
+    GIT_TAG        3634f2ded19e0cf38208c8b86cea9e1d7c8e397d # v0.25.0
+  )
+
+  FetchContent_MakeAvailable(fmt spdlog xtl xtensor)
+else()
+  find_package(spdlog)
+  find_package(xtensor)
+endif()
+
+## iree runtime
+
+if (NOT SHORTFIN_IREE_SOURCE_DIR AND SHORTFIN_BUNDLE_DEPS)
+  FetchContent_Declare(
+    iree
+    GIT_REPOSITORY https://github.com/iree-org/iree.git
+    GIT_TAG candidate-20240821.992
+    # TODO: We shouldn't have to pull googletest when we are not building tests.
+    #       This needs to be fixed with IREE.
+    GIT_SUBMODULES "third_party/benchmark third_party/cpuinfo third_party/flatcc third_party/hip-build-deps third_party/googletest"
```

Comment:
Interesting. I would have expected that this was semicolon delimitted or not quoted. I expect this is working like this as a quirk of passthrough to the underlying git command.

Reply:
FYI, looking into what's actually cloned, the submodules are fully fetched and `GIT_SHALLOW ON` does not seem to have an effect here. Anyway, this is probably worth a look if there someday is a need to build the compiler and fetch LLVM.

Reply:
Interesting, I think some options don't work as expected. I'll remove GIT_SHALLOW if it's a no-op anyway.

Reply:
It works for IREE but not for its submodules. So no need to remove it.